library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -410,13 +451,17 @@ private void handleCtrl(byte cc1, byte cc2, boolean repeatedControlPossible) {
       }
     }
 
+    if (!isBytePairForSelectedChannel(cc1)) {
+      return;
+    }
+
     if (isMidrowCtrlCode(cc1, cc2)) {
       handleMidrowCtrl(cc2);
     } else if (isPreambleAddressCode(cc1, cc2)) {
       handlePreambleAddressCode(cc1, cc2);
     } else if (isTabCtrlCode(cc1, cc2)) {
       currentCueBuilder.tabOffset = cc2 - 0x20;
-    } else if (isMiscCode(cc1, cc2)) {
+    } else if (isMiscCode(cc1, cc2, selectedField)) {
```

Comment:
Is it really necessary to check the field here? I acknowledge that the field will be set in `cc1` in this case, but wont the header check (`if ((ccHeader & CC_FIELD_FLAG) != selectedField) {`)  near the top of `decode` have already skipped the data if it belonged to a different field? If not then something seems a bit odd about this?

Reply:
This is weird in the standard, but the Misc code (only) bytes were defined to contain one additional bit based on field value. That bit seems to be redundant information, but needs to be checked, otherwise some rare values would satisfy the condition that should not be handled as Misc Code.

Reply:
This still seems odd to me. Either it's completely redundant information, in which case the check is not necessary (even though the field bit is present), or it's not redundant, in which case the header field and the misc code field can differ for a legitimate reason.

The way this code is currently written, we'll always ignore misc codes if the fields differ, either because the header specifies the field we want but the misc code doesn't, or because the header field specifies the field we don't want and we never get as far as checking the misc code field bit. If this is correct behavior, then presumably such data would never be inserted into the stream in the first place.

The only logical possibilities I can see are:
1. The field is completely redundant and the check is not necessary.
1. We should be handling misc codes if the field bit matches, even if the header specifies another field.
Neither of which is what this code is currently doing.

Reply:
What if we consider the other direction:
`boolean isMiscCode(cc1, cc2)`
cannot be a function deciding if the received byte pair CC1, CC2 is a Misc Code or not, as the byte values considered being a Misc Code vary based on the field value, too. So we need the additional third parameter to have a single function deciding if a byte pair is a Misc Code.

Without the third parameter the function might return false positives.

Reply:
I don't think that addresses my comment. I understand that there _exists_ a field bit, but the statement that it's therefore necessary to check it isn't true in possibility (1) above, because you already know what the field bit will be set to. In possibility (2) it is necessary to check the field bit, but that would imply we need to be making changes elsewhere to handle misc codes where the header has specified another field. Which of (1) and (2) is true?

Reply:
Let me consult the CEA608 Standard

Reply:
Did you figure anything out? If stuck on this I'd propose we proceed on the assumption that (1) is true (since it's easier). If we discover otherwise later on, we can fix in a subsequent change. Would you be OK with that? If so we'll get this merged (no need to change anything on your side; we can drop the check as part of the merge).

Reply:
I still do not understand how (1) can work, as it cannot be redundant as you need CC1, CC2, and the field value at the same time to figure out if a Byte pair is a misc code or not. Until we did not handle Field 2 and channel 2, this was not an issue, the original code only worked for Field 1 channel 1 correctly. It dropped anything that was Field 2 so this issue was not present.

I tried to look up if you can get a Field 2 misc code while processing byes from field 1, but there is no mention for that in the standard, so I would expect that (2) is also not valid.

    8.4 Closed Caption Mode When closed captioning is used on line 21, field 2, it shall conform to all of the applicable specifications and recommended practices as defined for field 1 services with the following differences:
    a)   The non-printing character of the miscellaneous control-character pairs that fall in the range of 0x14, 0x20 to 0x14, 0x2F in field 1, shall be replaced with 0x15, 0x20 to 0x15, 0x2F when used in field 2. 
    b)   The non-printing character of the miscellaneous control-character pairs that fall in the range of 0x1C, 0x20 to 0x1C, 0x2F in field 1, shall be replaced with 0x1D, 0x20 to 0x1D, 0x2F when used in field 2.
    c)   When any of the control codes from 0x01 to 0x0F is used to begin a control-character pair it indicates the beginning of XDS Data. This XDS data should be considered part of a different data channel and can be used to interrupt the captioning data according to the FCC rules for other data channels.

So, when we are processing field 2 byte pairs, we need to add this check for every control code processing. If that flag is not set, the value is not a valid Field 2 Control code, so we should drop it (at least from control code processing). We could handle the value of the flag in handleContorlCode() as well, but probably it is easier to handle it in isMiscCode().

Reply:
> it cannot be redundant

If we only had the two CC bytes (`cc_data_1` and `cc_data_2`) then this would be correct. But that's not all we have. We also have `ccHeader`, and we've already done a field check using it before we've got to this point. If the same thing is checked twice, then how is that not redundant?

As part of looking at https://github.com/google/ExoPlayer/issues/5656 I coincidentally discovered that this specific check is causing all captions to be discarded for CC3 for that piece of content. It seems the content has the field as expected in `ccHeader`, but not in the misc codes, resulting in them being dropped. This content works correctly in other players, suggesting that they're not checking the misc codes in this way.

For this reason I think we will proceed assuming (1). If you can find some content that's explicitly broken by doing this, please provide a sample and we can revisit. Thanks!

Reply:
To comply to section 8.4 of the CEA608 specification I quoted above, bytes should be handled as MISC control codes only if they start with 0x15 instead of 0x14 (for frame 2, so it means for channel 3 and 4).

If we do not check it here, do you suggest adding this to the function HandleMiscCode(CC2) then?

Note: Frame 2 (channel 3 and 4) and channel 2 used to be dropped completely, so this was not materializing as an issue before.

I have found these bugs while playing local Over-The-Air content in the Sunnyvale area. Do you have access to live content here?

Reply:
> To comply to section 8.4 of the CEA608 specification I quoted above, byte pairs should be handled as MISC control codes only if they start with 0x15 instead of 0x14 (for frame 2, so it means for channel 3 and 4).

The part you quoted doesn't say anything about how the client should behave.

I feel like we're going round in circles. I've been repeatedly asking a very specific question: There's _already_ a check that we're dealing with data for `selectedField` before this point in code, which uses `ccHeader`. The question I'm asking is why this check would be any different. If the checks are different, that would imply that a misc code for one field was given a `ccHeader` claiming it's actually for a different field. Why would that make sense? The right way to unblock this discussion is to address this specific question.

Reply:
(1) is not true, it is not redundant. When we are on Frame 2 (channel 3 or 4), the current code is checking for 0x14 and 0x1F for CC1 to be the first byte of the pair. 

(2) is not possible, I have tested many streams, but never seen a MISC code on Frame "A" addressing frame "B".

I simply added that check on line 464 because MISC codes on Frame 2 are defined to have the first byte 0x15 or 0x1D. Without also checking the frame value I could not decide when the original 0x14 and 0x1C should not be handled as MISC codes. Is  this clear?

I am OK if you go ahead assuming (1), and we get back to this later.

Reply:
You still have not explicitly addressed how the `ccHeader` check is materially different, which is a requirement for checking the field in the MISC code to not be redundant. Simply stating it's not redundant, again, without explaining when it would be expected to give a different result to the `ccHeader` check, does not address the question.

We will proceed assuming (1) for now. Thanks.